### ISSUE TITLE
RTX 3080ti issue with Alienware m15 R3

### ISSUE THREAD

**Issue by jimblust** [Shidell/OpenCaldera#8]
I have an Alienware m15 R3 that I’m trying to get working with my Alienware Graphics Amplifier (upgraded with a Corsair CX650M power supply) and an RTX 3080ti. My m15 R3 has an onboard RTX 2070 Super and Intel UHD 630 Graphics, 300Hz non-gsync display, and latest BIOS 1.8.0. I’m using the Nvidia 471.11 notebook driver.

I’ve tried various things. After running DDU, installing 471.11 without AGA to get the 2070 Super running, rebooting to connect the AGA, I start out with the 2070 operational and the 3080 showing up as a “Microsoft display adapter” with a code 32. Then, I install the 471.11 again, and the 3080ti display fires up with a mirror of the laptop display for a couple of seconds during the driver install, and then the external screen freezes. After the install is complete, 3080ti shows up with a code 43 in device manager, and the 2070 is operational. After a reboot, it takes a few seconds of a black screen before the windows “device disconnected” sound , and after logging in, both the 3080ti and 2070 have a code 43. (Note: This is a m15 R3 with the 300Hz screen, so I don’t ever see the Intel adapter in device manager.) 

Once, with HackFlags set to 200, I was able to log in, both displays working, and I could see the 3080ti, 2070, and Intel listed as working in Device Manager. It lasted less than 30 seconds before the external display froze, and 3080ti and 2070 went back to code 43 in device manager. This often coincides with a crash and restart of dwm.exe (the desktop window manager) in the event viewer. I seem to get similar results without HackFlags, or with it set to 200, 400, or 600. Sometimes, I can get the external display working again for a few seconds by disabling the 3080ti and enabling it again in Device manager, but then it freezes again.

I suspect that my specific hardware configuration on this model is something that’s still not handled correctly by Nvidia’s drivers

Other configurations that do work:

1. The AGA and RTX 3080 ti work fine with my older Alienware 15 laptop, which has on onboard GTX 970M and Intel HD Graphics 4600, with the 471.11 notebook driver and no tweaks , so I know the AGA and 3080ti and CX650M are working.

2. The Alienware m15 R3 works with the AGA with my old GTX 980ti installed and no HackFlags needed, so I don’t think there’s any total incompatibility between the m15 R3 and the AGA.

**Comment by Shidell** [comment]
Do you have a hardware (or software) MUX switch where you can disable the 2070? I'm curious if that's an option to try (WRT the 3080Ti.)

**Comment by jimblust** [comment]
No, I did find this thread referencing Fn-F7 for the mux on the m15 R3 with the 144Hz display, but that key doesn't isn't labeled for a Fn function on my keyboard with the 300Hz display and pressing it doesn't do anything (including after installing the Alienware On Screen Display app).

https://www.dell.com/community/Alienware/m15-R3-using-MUX-switch-to-integrated-graphics/td-p/7650019

**Comment by ALEX8642** [comment]
I have the same setup and issue with an m15r3 and TUF 3090 in the AGA. I actually get different codes sometimes though. 31 with or without hackflags, code 12 sometimes... Haven't seen 43 though. Setting hackflags to 600 just makes my system crash too BSOD (DRIVER DMA VIOLATION). I'd really like to use the AGA instead of having to use a TB3 enclosure device. Oh and every time I try this it breaks my integrated 2070 (code 31 without AGA connected after restart) and I have to delete and reinstall its drivers to get it working again. EDIT: Was playing with it again this morning and now windows won't even boot anymore.

**Comment by Shidell** [comment]
@ALEX8642 Are you using Win11? The latest Nvidia drivers? Is your firmware up-to-date?

Assuming everything is up-to-date, if you don't have a MUX switch (physical and/or software/BIOS), I'm not sure what might be causing the issues you're seeing. Do you have a custom PSU in your AGA? The stock PSU isn't sufficient for a 3090 and might be causing random errors/failures.

**Comment by ALEX8642** [comment]
Hi @Shidell Alienware m15r3 (i9-11980, RTX 2070S, but its basically garbage because it has never been able to produce a stable framerate even after a repaste and custom tuning/undervolting. Of course Alienware did not consider that enough for warranty, but that is another story)... That was one reason I wanted this to work... Going back to the original issue though (apologies) everything was up to date. The issue presented itself with and without custom PSU. I ended up reinstalling windows as it was unrecoverable - which is crazy (Yes Win 11). In any case, it will be cool if this ever works but I got my main rig back up and running again so the 3090 is back in its rightful place (had a bad CPU of all things). I'll probably never plug the AGA back in again for fear of this issue recurring.

**Comment by Shidell** [comment]
@ALEX8642 I do believe one other individual with an R3 had problems getting this solution to work; unfortunately, without having access to an R3, there's little I can do in terms of troubleshooting--if I could get my hands on one, I'd love to do so, but as you can imagine, it isn't very feasible short of buying one for myself, which isn't practical.

With respect to the CPU/GPU and framerate/temps, I've found IC Diamond produces the best results, because it's a thicker, more viscous compound, which does a better job of bridging the gap between the HSF and the dies. I recommend laptops use it and be liberal in their application to ensure there's sufficient thermal compound (as too much is OK, but too little is not.)

So you've tried HackFlags 200 & 600 and neither work whatsoever? And you tried installing the full Nvidia GeForce driver suite?

**Comment by ALEX8642** [comment]
@Shidell Correct. Hackflags 600 of course being the one that somehow caused total windows failure. With hackflags 200 it would switch to intel graphics and throw a code 12 or 31 for the laptops 2070S (this includes the regular reinstall while the 3090 was still plugged in - plus several reboots - code would not go away with or without the 3090), requiring a complete scrubbing of Nvidia drivers (DDU) and then full reinstall to recover and bring the 2070S back to normal ok status.

**Comment by Shidell** [comment]
@ALEX8642 Have you checked for an enable/disable toggle in the firmware? Does your model have a physical MUX switch?
